lego/apps/email/notifications.py

```diff
@@ -0,0 +1,101 @@
+from datetime import timedelta
+
+from django.utils import timezone
+
+from lego.apps.events.constants import EVENT_TYPE_TRANSLATIONS
+from lego.apps.events.models import Event
+from lego.apps.joblistings.constants import JOB_TYPE_TRANSLATIONS
+from lego.apps.joblistings.models import Joblisting
+from lego.apps.notifications.constants import WEEKLY_MAIL
+from lego.apps.notifications.notification import Notification
+from lego.apps.permissions.constants import VIEW
+from lego.apps.tags.models import Tag
+
+
+class WeeklyNotification(Notification):
+
+    name = WEEKLY_MAIL
+
+    def generate_mail(self):
+        yesterday_timestamp = timezone.now() - timedelta(days=1)
+        last_sunday_timestamp = timezone.now() - timedelta(days=7)
+
+        weekly_tag = Tag.objects.filter(tag="weekly").first()
+        # Check if weekly tag exists so it does not crash if some idiot deletes the weekly tag
+        todays_weekly = (
+            weekly_tag.article_set.filter(created_at__gt=yesterday_timestamp).first()
+            if weekly_tag
+            else None
+        )
```

Comment:
Hmmm, I feel like this code is kinda relying on a lot of things going right, which could go wrong. F.ex. if someone creates the weekly article before `yesterday_timestamp` (also if this task doesn't run exactly at the right day).

Would prefer that this maybe include some more leeway

Reply:
As far as I know, it is always written on a sunday, but I have no problem with more leeway. Would three days be sufficient?